### ISSUE TITLE
problem with NaiveTSNE.py

### ISSUE THREAD

**Issue by overshiki** [CannyLab/tsne-cuda#3]
Hi,
I just want to test the performance of the version n^2 computational complexity(naive version as in original paper). However it seems currently the python wrapper for naive version is of some kind of problem. I fix the bug according to my understanding, as shown below. The code then worked for small datasets(15000by30 minist data), but the resulting graph is not correct. I am wondering maybe I made mistakes in debug. 

Another issue is that currently the n^2 naive version only support for small amount of data, it can not go through full minist dataset(60000by30, with PCA dim reduction from 28*28 to 30 dim).

Also, how could I set the specific gpu I want to run the code on?
 
```diff
diff --git a/src/python/tsnecuda/NaiveTSNE.py b/src/python/tsnecuda/NaiveTSNE.py
index a9b8b00..c060273 100644
--- a/src/python/tsnecuda/NaiveTSNE.py
+++ b/src/python/tsnecuda/NaiveTSNE.py
@@ -72,14 +72,14 @@ class NaiveTSNE(object):
         else:
             self.metric = metric
         self.verbose = int(verbose)
-        if random_seed is not None:
-            self.random_seed = int(random_seed)
-        else:
-            self.random_seed = os.urandom()
+        # if random_seed is not None:
+        #     self.random_seed = int(random_seed)
+        # else:
+        #     self.random_seed = os.urandom()
 
         # Build the hooks for the Naive T-SNE library
         self._path = pkg_resources.resource_filename('tsnecuda','') # Load from current location
-        self._lib = N.ctypeslib.load_library('libtsnecuda', _path) # Load the ctypes library
+        self._lib = N.ctypeslib.load_library('libtsnecuda', self._path) # Load the ctypes library
 
         # Hook the naive T-SNE function
         self._lib.pymodule_naive_tsne.restype = None
@@ -92,7 +92,7 @@ class NaiveTSNE(object):
                                   ctypes.c_float, # Learning Rate
                                   ctypes.c_int, # n_iter
                                   ctypes.c_int, # n_iter w/o progress
-                                  ctypres.c_float # min_norm
+                                  ctypes.c_float # min_norm
                                 ]
 
         # Set up the attributed
@@ -112,15 +112,15 @@ class NaiveTSNE(object):
 
         X = N.require(X, N.float32, ['F_CONTIGUOUS', 'ALIGNED'])
         self.embedding_ = N.empty(shape=(X.shape[0],self.n_components))
-        self.embedding_ = N.require(results, N.float32, ['F_CONTIGUOUS', 'ALIGNED', 'WRITEABLE'])
+        self.embedding_ = N.require(self.embedding_, N.float32, ['F_CONTIGUOUS', 'ALIGNED', 'WRITEABLE'])
         self._lib.pymodule_naive_tsne(X, self.embedding_, X.ctypes.shape, 
-                                        c_int(self.n_components), 
-                                        c_float(self.perplexity), 
-                                        c_float(self.early_exaggeration),
-                                        c_float(self.learning_rate), 
-                                        c_int(self.n_iter),
-                                        c_int(self.n_iter_without_progress),
-                                        c_float(self.min_grad_norm))
+                                        ctypes.c_int(self.n_components), 
+                                        ctypes.c_float(self.perplexity), 
+                                        ctypes.c_float(self.early_exaggeration),
+                                        ctypes.c_float(self.learning_rate), 
+                                        ctypes.c_int(self.n_iter),
+                                        ctypes.c_int(self.n_iter_without_progress),
+                                        ctypes.c_float(self.min_grad_norm))
         return self.embedding_
```

**Comment by DavidMChan** [comment]
Hello! Currently, the Naive method isn't really under development. We're working primarily on the BH-TSNE code. A more up to date version of the naive method can be found in the "Refactor" branch which should be significantly better than the naive in the master branch (but the python bindings may not actually work). We're working currently on version 1.0 of the code which will release a tested version of the Naive code, however we're tracking down an odd bug which occurs in some of the CUDA which is preventing us from releasing.

**Comment by overshiki** [comment]
Oh! thanks. 
Got it, I'll check that then.
Have already tested the speed of the BH version, about 100x faster than sklearn version on 1080Ti and intel Xeon E5. Really nice.

**Comment by DavidMChan** [comment]
For future reference, we've removed this capability from the code, and we won't be supporting a Naive version in this package.
